raiden-ts/src/transfers/epics/secret.ts

```diff
@@ -218,30 +294,154 @@ export const monitorSecretRegistryEpic = (
         // emits only if lock didn't expire yet
         secrethash in sent && sent[secrethash].transfer[1].lock.expiration.gte(blockNumber!),
     ),
-    map(([[secrethash, secret, event]]) =>
-      transferSecretRegistered(
-        {
-          secret,
-          txHash: event.transactionHash! as Hash,
-          txBlock: event.blockNumber!,
-          confirmed: undefined,
-        },
-        { secrethash },
-      ),
-    ),
+    mergeMap(function*([[secrethash, secret, event], { sent, received }]) {
+      if (
+        secrethash in sent &&
+        sent[secrethash].transfer[1].lock.expiration.gte(event.blockNumber!)
+      ) {
+        yield transferSecretRegister.success(
+          {
+            secret,
+            txHash: event.transactionHash! as Hash,
+            txBlock: event.blockNumber!,
+            confirmed: undefined,
+          },
+          { secrethash, direction: Direction.SENT },
+        );
+      }
+      if (
+        secrethash in received &&
+        received[secrethash].transfer[1].lock.expiration.gte(event.blockNumber!)
+      ) {
+        yield transferSecretRegister.success(
+          {
+            secret,
+            txHash: event.transactionHash! as Hash,
+            txBlock: event.blockNumber!,
+            confirmed: undefined,
+          },
+          { secrethash, direction: Direction.RECEIVED },
+        );
+      }
+    }),
   );
 
 /**
  * A simple epic to emit transfer.success when secret register is confirmed
  *
- * @param action$ - Observable of transferSecretRegistered actions
+ * @param action$ - Observable of transferSecretRegister.success actions
  * @returns Observable of transfer.success actions
  */
 export const transferSuccessOnSecretRegisteredEpic = (
   action$: Observable<RaidenAction>,
 ): Observable<transfer.success> =>
   action$.pipe(
-    filter(transferSecretRegistered.is),
+    filter(transferSecretRegister.success.is),
     filter(action => !!action.payload.confirmed),
     map(action => transfer.success({}, action.meta)),
   );
+
+/**
+ * Process newBlocks and pending received transfers. If we know the secret, and transfer doesn't
+ * get unlocked before revealTimeout blocks are left to lock expiration, request to register secret
+ *
+ * @param action$ - Observable of newBlock actions
+ * @param state$ - Observable of RaidenStates
+ * @returns Observable of transferSecretRegister.request actions
+ */
+export const transferAutoRegisterEpic = (
```

Comment:
Should we automatically register the secret of a transfer on chain, even if the amount is really low (even lower than gas to be paid)?

Reply:
There're some TODOs about economic viability of on-chain operations, we definitely must check on this later, but AFAIK, Raiden doesn't check either yet. I'll add a todo here for us to check it as well. Thanks for spotting this.

Reply:
In the best case this should be configurable per token. Because it depends on the market values of each token and they might be completely different

Reply:
Might be better to create follow up issues instead of comment todos